### DISCUSSION TITLE
Selfhosted upgrade supbase database Postgres from v14 to v15 constraint errors

### DISCUSSION THREAD

**Discussion by devidevio** [supabase/supabase#12626] [Questions]
I have a problem that is causing me a lot of headaches. I'm not able to create a backup (dump) of my database (Postgres v14) and restore / recreate it on a completely new database (Postgres v15).

The problem at this point is, that the constraints are checked while importing (inserting) the data. As I rely a lot on the auth- and storage helper functions (f. e. `auth.uid()`) I'm not able to skip these checks.

I have a lot of complex and separately linked functions for authentication and authorization which are not "skipped" / ignored by the database superuser (whether `postgres` nor `supabase_admin`). I tried every possible solution I found in any documentation / forum, especially:
* Setting the `--disable-triggers` flag on `pg_restore` with `--superuser=[USER]` set.
* Setting `SET session_replication_role = replica;` at the top of the created database dump (seems to be ignored?)
* Opening a bash for my database docker container and setting the `SET session_replication_role = replica;` and running the restore command.
* Disabling all triggers on all tables before creating the database dump.
* Partially returning just `TRUE` in my functions for authentication / authorization (no effect).

Currently it's not possible for me to delete all the constraint (checks) as they are spread everywhere across my tables / functions.

Does anyone have a solution for this case?

When I try to start a new database (Postgres v15) container on the old database data (Postgres v14) it crashes immediately that the data between v14 and v15 is not compatible).

**Comment by ruggi99** [comment]
Have you tried with `pg_upgrade`?
Of course it's not so easy when postgresql is used in docker containers.

Major versions are not compatible, so postgresql 15 refuses to start on v14 data, it's normal.

Ora you can create a docker container (don't know if it exists already) where you install both v14 and v15 and you migrate the data in the attached volume. Might be a new project to develop
